mesonbuild/interpreter.py

```diff
@@ -2830,6 +2878,56 @@ def func_message(self, node, args, kwargs):
     def message_impl(self, argstr):
         mlog.log(mlog.bold('Message:'), argstr)
 
+    @noArgsFlattening
+    @permittedKwargs({'bool_yn'})
+    @FeatureNew('summary', '0.53.0')
+    def func_summary(self, node, args, kwargs):
+        if len(args) == 1:
+            if not isinstance(args[0], dict):
+                raise InterpreterException('Argument 1 must be a dictionary.')
+            section = ''
+            values = args[0]
+        elif len(args) == 2:
+            if not isinstance(args[0], str):
+                raise InterpreterException('Argument 1 must be a string.')
+            if isinstance(args[1], dict):
+                section, values = args
+            else:
+                section = ''
+                values = {args[0]: args[1]}
+        elif len(args) == 3:
+            if not isinstance(args[0], str):
+                raise InterpreterException('Argument 1 must be a string.')
```

Comment:
Maybe say "... a dictionary or a string" in all three cases. Having not enough arguments by mistake is also possible, and then the message will be confusing.

Reply:
Hmm, I think it will be confusing both ways, we don't know if user screwed the number of args or the type of args. I think it's less likely that he made a mistake in the number of args. I think if someone is confused, then he needs to read the doc.